app/forms/draft_work_form.rb

```diff
@@ -158,6 +158,19 @@ def description
     super
   end
 
+  # the terms agreement default to saved value for this object if persisted,
+  #  or false if (a) never accepted or (b) not accepted in the last year,
+  #  else it defaults to true
+  #  (i.e a new draft with a previously agreed work in the last year defaults to true)
+  def default_agree_to_terms
+    return model.fetch(:work_version).agree_to_terms if persisted?
+
+    last_work_terms_agreement = model.fetch(:work).collection.creator.last_work_terms_agreement
```

Comment:
why look at the collection creator and not the work depositor?

Reply:
yes, and in fact this is wrong because the collection creator could be different than the work depositor...

the problem I am running into is that that the work depositor does not appear to be set in the work model for a brand new work until it is persisted:

```
(byebug) model.fetch(:work_version).work
#<Work id: nil, druid: nil, head_id: nil, collection_id: 1, depositor_id: nil, created_at: nil, updated_at: nil>
```

I also don't appear to have access to the current logged in user (`current_user`) in this class ... any thoughts on best way to get the current access to the current depositor from here?

Reply:
You can do like this:
https://github.com/sul-dlss/happy-heron/blob/main/app/controllers/works_controller.rb#L23

on this line:
https://github.com/sul-dlss/happy-heron/blob/main/app/controllers/works_controller.rb#L14